src/main/java/it/gov/pagopa/pu/send/repository/SendNotificationRepositoryExtImpl.java

```diff
@@ -0,0 +1,40 @@
+package it.gov.pagopa.pu.send.repository;
+
+import it.gov.pagopa.pu.send.enums.NotificationStatus;
+import it.gov.pagopa.pu.send.model.SendNotification;
+import it.gov.pagopa.pu.send.model.SendNotification.Fields;
+import org.springframework.data.mongodb.core.FindAndModifyOptions;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class SendNotificationRepositoryExtImpl implements SendNotificationRepositoryExt {
+
+  private final MongoTemplate mongoTemplate;
+
+  public SendNotificationRepositoryExtImpl(MongoTemplate mongoTemplate) {
+    this.mongoTemplate = mongoTemplate;
+  }
+
+  @Override
+  public SendNotification createIfNotExists(Long sendNotificationId, SendNotification sendNotification) {
+    return mongoTemplate.findAndModify(
+      Query.query(Criteria.where(Fields.sendNotificationId).is(sendNotification.getSendNotificationId())
+      .and(Fields.preloadId).is(sendNotification.getPreloadId())),
```

Comment:
why don't you create a UNIQUE index in order to fail if trying to store it? this way you don't event need a custom query, you could just use the SpringDataRepository interface